internal/api/v1beta1/notification.go

```diff
@@ -94,3 +97,42 @@ func (s *GRPCServer) ListNotificationMessages(ctx context.Context, req *sirenv1b
 		Messages: items,
 	}, nil
 }
+
+func (s *GRPCServer) ListNotifications(ctx context.Context, req *sirenv1beta1.ListNotificationsRequest) (*sirenv1beta1.ListNotificationsResponse, error) {
+	notifications, err := s.notificationService.List(ctx, notification.Filter{
+		Type:             req.GetType(),
+		Template:         req.GetTemplate(),
+		Labels:           req.GetLabels(),
+		ReceiverSelector: req.GetReceiverSelectors(),
+	})
+	if err != nil {
+		return nil, s.generateRPCErr(err)
+	}
+
+	items := []*sirenv1beta1.Notification{}
+
+	for _, notification := range notifications {
+		data, err := structpb.NewStruct(notification.Data)
+		if err != nil {
+			return nil, s.generateRPCErr(err)
+		}
+
+		currentTime := time.Now()
+		validDuration := currentTime.Add(notification.ValidDuration)
```

Comment:
why do we need to do this?

Reply:
In the proto ValidDuration is expecting `*timestamppb.Timestamp`  but we are getting `time.Duration`. 
That is reason for conversion but  need to make change in proto and check if it supports `time.Duration` type.

Reply:
Then the proto is wrong right? This valid duration is not a timestamp

Reply:
Yes @mabdh